core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpOpenLineageDispatcher.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.dispatcher
+
+import org.apache.commons.configuration.Configuration
+import org.apache.spark.internal.Logging
+import scalaj.http.{Http, HttpStatusException}
+import za.co.absa.commons.version.Version
+import za.co.absa.spline.harvester.dispatcher.httpdispatcher.rest.{RestClient, RestEndpoint}
+import za.co.absa.spline.harvester.dispatcher.modelmapper.{ModelMapper, OpenLineageModelMapper}
+import za.co.absa.spline.harvester.dispatcher.openlindispatcher.{HttpOpenLineageDispatcherConfig, RESTResource}
+import za.co.absa.spline.producer.model.{ExecutionEvent, ExecutionPlan}
+
+import javax.ws.rs.core.MediaType
+import scala.util.control.NonFatal
+
+
+class HttpOpenLineageDispatcher(restClient: RestClient, apiVersion: Version, openLineageNamespace: String)
+  extends LineageDispatcher
+    with Logging {
+
+  import za.co.absa.spline.harvester.json.HarvesterJsonSerDe.impl._
+
+  def this(dispatcherConfig: HttpOpenLineageDispatcherConfig) = this(
+    HttpOpenLineageDispatcher.createDefaultRestClient(dispatcherConfig),
+    dispatcherConfig.apiVersion,
+    dispatcherConfig.namespace
+  )
+
+  def this(configuration: Configuration) = this(new HttpOpenLineageDispatcherConfig(configuration))
+
+
+  override def name = "Open Lineage Http"
+
+  logInfo(s"Using Producer API version: ${apiVersion.asString}")
+
+  private val lineageEndpoint = restClient.endpoint(RESTResource.Lineage)
+  private val modelMapper = ModelMapper.forApiVersion(apiVersion)
+  private val openLineageModelMapper = new OpenLineageModelMapper(modelMapper, apiVersion, openLineageNamespace)
+
+  private var cachedPlan: ExecutionPlan = null
```

Comment:
Unfortunately in Scala keywords `val` and `var` are dangerously similar in look :)
I encourage making mutable variables as explicit as possible, especially in the project where a mutable variables occur very exceptionally. I would even prefer having some simple FSM here to isolate mutable stuff as much as possible from the rest of the code. But if not FSM then at least let's name it with underscore in front, following Scala mutable properties naming convention. I also sometimes like to annotate it with `@volatile`. Yes, it's completely redundant, and I'm not insisting on it, but for me it helps to draw that additional bit of attention and screams to the reader that the variable is _mutable_.

```suggestion
  @volatile
  private var _cachedPlan: ExecutionPlan = _
```

Reply:
FSM == finite state machine, right?

underscore - ok that could work
We could also refactor the dispatcher interface

@volatile -  not a fan of this - volatile has its own meaning and using it as a marker for something else seems strange, why not just create @Mutable annotation in commons?